### PR TITLE
Add filesystem catalog docs + deprecate grpc services

### PR DESCRIPTION
**Pull request by bmaidics**
Depends on https://github.com/aklivity/zilla/pull/962

### CODE REVIEW

File: src/reference/config/catalogs/catalog-filesystem.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+description: Zilla runtime filesystem catalog
+category:
+  - Catalog
+---
+
+# filesystem Catalog
+
+::: important Feature is in Incubator
+Read how to [enable incubator features](../../../how-tos/install.md#enable-incubator-features). Star and watch the [Zilla repo](https://github.com/aklivity/zilla/releases) for new releases!
+:::
```
Comment (vordimous) [inline src/reference/config/catalogs/catalog-filesystem.md:12]:
```suggestion
::: important Feature is in Incubator
Read how to [enable incubator features](../../../how-tos/deploy-operate.md#enable-incubator-features). Star and watch the [Zilla repo](https://github.com/aklivity/zilla/releases) for new releases!
:::
```

File: src/reference/config/bindings/binding-grpc.md [hunk 2]
```diff
@@ -81,12 +81,13 @@ options:
...
 ```
 
-#### options.services
+#### options.services (deprecated)
 
 > `array` of `string`
 
 Protobuf service definition filenames, typically with `.proto` filename extension.
 
+Note: this option is deprecated. Use filesystem catalog instead for accessing `.proto` files from the filesystem.
```
Comment (vordimous) [inline src/reference/config/bindings/binding-grpc.md:90]:
```suggestion
Note: this option is deprecated. Use filesystem catalog instead for accessing `.proto` files from the filesystem.

```

File: src/reference/config/bindings/binding-grpc.md [hunk 3]
```diff
@@ -43,7 +43,7 @@ The `client` kind `grpc` binding adapts `grpc` request-response streams to `http
 
 - [kind\*](#kind)
 - [options](#options)
-  - [options.services](#options-services)
+  - [options.services](#options-services) (deprecated)
```
Comment (vordimous) [inline src/reference/config/bindings/binding-grpc.md:46]:
```suggestion
  - [options.services](#options-services)
```

File: src/reference/config/bindings/binding-grpc.md [hunk 4]
```diff
@@ -81,12 +81,14 @@ options:
     - proto/echo.proto
 ```
 
-#### options.services
+#### options.services (deprecated)
```
Comment (vordimous) [inline src/reference/config/bindings/binding-grpc.md:84]:
```suggestion
#### options.services

::: warning Deprecated
This property will be removed in a future release.
:::
```

File: src/reference/config/catalogs/catalog-filesystem.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
+---
+shortTitle: filesystem
+description: Zilla runtime filesystem catalog
+category:
+  - Catalog
+---
+
+# filesystem Catalog
+
+::: important Feature is in Incubator
```
Comment (vordimous) [inline src/reference/config/catalogs/catalog-filesystem.md:10]: is this catalog going to be in incubator or enabled in the full release?

File: src/reference/config/catalogs/catalog-filesystem.md [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+Zilla runtime filesystem catalog
+
+```yaml {2}
+catalogs:
+  catalog0:
+    type: filesystem
+    options:
+      subjects:
+        petstore:
+          url: schema.yaml
```
Comment (vordimous) [inline src/reference/config/catalogs/catalog-filesystem.md:23]: Better to use more reader-friendly names
```suggestion
catalogs:
  my_catalog:
    type: filesystem
    options:
      subjects:
        my_local_file:
          url: path/to/local_file.txt
```

File: src/reference/config/catalogs/catalog-filesystem.md [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+
+> `object`
+
+#### options.subjects\*
+
+> `object` as map of named objects
+
+Unique identifier for artifact categorization in the catalog.
+
+##### subjects.url\*
```
Comment (vordimous) [inline src/reference/config/catalogs/catalog-filesystem.md:55]: If this is a local or remote location then `url` is probably fine, but if we are going to limit it to the host filesystem then this should be `path` or `relative-path`?

File: src/reference/config/catalogs/catalog-filesystem.md [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+
+> `string`
+
+URL to access artifact from the filesystem. The URLs are relative to `zilla.yaml`.
+
+##### subjects.version
```
Comment (vordimous) [inline src/reference/config/catalogs/catalog-filesystem.md:61]: How does the version concept work for the host filesystem?

**Review by vordimous — CHANGES_REQUESTED** [review]

File: src/reference/config/catalogs/catalog-filesystem.md (hunk 5 above)
Comment (bmaidics) [reply src/reference/config/catalogs/catalog-filesystem.md:10]: @jfallows I think this question is yours.

File: src/reference/config/catalogs/catalog-filesystem.md (hunk 7 above)
Comment (bmaidics) [reply src/reference/config/catalogs/catalog-filesystem.md:55]: Right now readURL supports remote location as well. @jfallows , what do you think?

File: src/reference/config/bindings/binding-grpc.md [hunk 9]
```diff
@@ -83,10 +83,16 @@ options:
...
 
+::: warning Deprecated
+This property will be removed in a future release.
+:::
+
 > `array` of `string`
 
 Protobuf service definition filenames, typically with `.proto` filename extension.
 
+Note: this option is deprecated. Use filesystem catalog instead for accessing `.proto` files from the filesystem.
```
Comment (jfallows) [inline src/reference/config/bindings/binding-grpc.md:94]: Recommend we make `filesystem catalog` also link to the reference doc for `catalog-filesystem.md`.

File: src/reference/config/catalogs/catalog-filesystem.md (hunk 7 above)
Comment (jfallows) [reply src/reference/config/catalogs/catalog-filesystem.md:54]: This should be `path` for `filesystem` catalog.

Internally, `readURL` is scheduled for removal.

**Review by jfallows — CHANGES_REQUESTED** [review]

File: src/reference/config/bindings/binding-grpc.md (hunk 9 above)
Comment (vordimous) [inline src/reference/config/bindings/binding-grpc.md:94]:
```suggestion
```

File: src/reference/config/bindings/binding-grpc.md [hunk 10]
```diff
@@ -83,10 +83,16 @@ options:
 
 #### options.services
 
+::: warning Deprecated
+This property will be removed in a future release.
```
Comment (vordimous) [inline src/reference/config/bindings/binding-grpc.md:87]:
```suggestion
This property will be removed in a future release. To access' .proto' files from the filesystem, use the [filesystem](../catalogs/catalog-filesystem.md) catalog instead.
```

File: src/reference/config/bindings/binding-grpc.md [hunk 11]
```diff
@@ -83,10 +83,15 @@ options:
...
 
+::: warning Deprecated
+This property will be removed in a future release. To access '.proto' files from the filesystem, use the [filesystem](../catalogs/catalog-filesystem.md) catalog instead.
+:::
+
 > `array` of `string`
 
 Protobuf service definition filenames, typically with `.proto` filename extension.
 
+
```
Comment (vordimous) [inline src/reference/config/bindings/binding-grpc.md:94]:
```suggestion
```

**Review by vordimous — APPROVED** [review]

**Review by jfallows — APPROVED** [review]

**Pull request merged**